### PR TITLE
feat: custom http client

### PR DESCRIPTION
**Pull request by Vinzent03**

### CODE REVIEW

**Review by dshukertjr — COMMENTED** [review]
@Vinzent03 
Nice one! Could we also just include the package update to 1.0.0-dev.2 in this PR as well?

**Comment by Vinzent03** [comment]
sure

File: test/widget_test_stubs.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ class MockWidget extends StatefulWidget {
   const MockWidget({Key? key}) : super(key: key);
 
   @override
-  _MockWidgetState createState() => _MockWidgetState();
+  State<MockWidget> createState() => _MockWidgetState();
```
Comment (dshukertjr) [inline test/widget_test_stubs.dart:10]: @Vinzent03 
What is this change for BTW?

Comment (bdlukaa) [reply test/widget_test_stubs.dart:10]: https://dart-lang.github.io/linter/lints/library_private_types_in_public_api.html

Comment (Vinzent03) [reply test/widget_test_stubs.dart:10]: It's a lint from the new `lints` package. I had to replace it many times in my application with the lint switch.

Comment (dshukertjr) [reply test/widget_test_stubs.dart:10]: Ah, nice! Wasn't aware of that!

**Review by dshukertjr — APPROVED** [review]
Amazing! Thank you so much for completing such an amazing feature!

**Pull request merged**